content/chapters/software-stack/lecture/slides/software-stack.md

```diff
@@ -0,0 +1,133 @@
+---
+
+## Software Stack
+
+---
+
+![What We Have](media/hardware-software/1-hardware.svg)
+
+----
+
+![What We Want](media/hardware-software/2-features.svg)
+
+----
+
+![Have to Want](media/hardware-software/3-software.svg)
+
+----
+
+![Hardware + Software](media/software-use.svg)
+
+----
+
+### Flexibility of Software
+
+* Hardware provides basic actions
+<!-- .element: class="fragment" -->
+* Users request rich features
+<!-- .element: class="fragment" -->
+* Software bridges the gap between user demands and hardware possibility
+<!-- .element: class="fragment" -->
+
+----
+
+### Hardware vs Software
+
+| Hardware                         | Software                                   |
+| :------------------------------: | :----------------------------------------: |
+| do one thing, do one thing well  | flexible                                   |
+| efficient                        | featureful                                 |
+| unmodifiable                     | (un)installable, configurable              |
+| physical                         | virtual, easy to duplicate                 |
+| use once at a time               | reusable                                   |
+| monolithic                       | (de/re)composable                          |
+
+---
+
+### Properties of Software
+
+* Reusability
+* Portability
+* Configurability / Flexibility
+
+----
+
+### Reusability
+
+<img src="media/software-reuse.png" alt="Reusability" width="60%" />
+<!-- https://medium.com/application-library-engineering-group/a-deep-dive-into-reusable-components-3788ca756390 -->
+
+----
+
+![Don't Reinvent the Wheel](media/dont-reinvent-the-wheel.jpeg)
+<!-- https://levelup.gitconnected.com/software-engineering-best-practices-for-reusing-in-software-development-6006f9d8d364 -->
+
+----
+
+### Portability
+
+![Portability](media/portability.svg)
+
+----
+
+### Configurability
+
+Same software, different use cases
+
+![Software Configurability](media/software-configurability.svg)
+
+---
+
+### Interface and Implementation
+
+![Inteface and Implementation](media/interface-implementation.svg)
```

Comment:
The logos of Firefox and Vim aren't rendered. Export it using the method I described in my comment to fix this.

Reply:
They are rendered on my side.